src/generators/schema/faq.php

```diff
@@ -72,12 +72,17 @@ public function is_needed( Meta_Tags_Context $context ) {
 	public function generate( Meta_Tags_Context $context ) {
 		$ids   = [];
 		$graph = [];
-		foreach ( $context->blocks['yoast/faq-block']['attrs']['questions'] as $index => $question ) {
-			if ( ! isset( $question['jsonAnswer'] ) || empty( $question['jsonAnswer'] ) ) {
-				continue;
+		if ( isset( $context->blocks['yoast/faq-block']['attrs']['questions'] ) ) {
+			foreach ( $context->blocks['yoast/faq-block']['attrs']['questions'] as $index => $question ) {
+				if ( ! isset( $question['jsonAnswer'] ) || empty( $question['jsonAnswer'] ) ) {
+					continue;
+				}
+				$ids[]   = [ '@id' => $context->canonical . '#' . $question['id'] ];
+				$graph[] = $this->generate_question_block( $question, $index, $context );
 			}
-			$ids[]   = [ '@id' => $context->canonical . '#' . $question['id'] ];
-			$graph[] = $this->generate_question_block( $question, $index, $context );
+		}
+		else {
+			return $graph;
```

Comment:
The code could be simpler (less indented).

If you make the first if-statement a negation...
```
if ( ! isset( $context->blocks['yoast/faq-block']['attrs']['questions'] ) ) {
    return $graph;
}
```

... then, the rest of the original code can follow without being indented (and the `else` clause can disappear).

Reply:
The FAQ-block does not seem to throw an error when it is empty. So this is not relative anymore.